modules/ROOT/partials/configuration/dialog_streamContent.adoc

```diff
@@ -0,0 +1,58 @@
+[[streamContent]]
+== `streamContent`
+
+`+streamContent+` is an optional property, available to add to the `+iframe+` dialog component.
+
+The property has a default value of `+false+`.
+
+When `+streamContent+` is set to true, content presented within the `+iframe+` dialog component is updated using the `document.write()` method, rather than the by using the `+srcdoc+` attribute.
+
+The `+srcdoc+` attribute requires the `+iframe+` be reloaded each time the content changes. This results in a flickering effect.
+
+As well, performing updates via the `+srcdoc+` attribute means the contents of the `+iframe+` are not accessible. As a consequence, it is not possible to automatically scroll the `+iframe+` contents to the bottom as it gets updated.
+
+Setting `+streamContent+` to `+true+` switches the update mechanism to the `document.write()` method and presents incoming content as a stream at an end-of-the-current data position.
```

Comment:
Not really sure what you mean by "presents incoming content as a stream at an end-of-the-current data position". 

Setting `streamContent` to `true` to update content would:
1. Minimise flickering by using `document.write()` as an optimisation for updating content rapidly when streaming, and
2. Keep the scroll at the bottom of the iframe document _if before the update the scroll was already at the bottom_. The part in italics is important as this would allow for the user to freely scroll around the iframe as it is being updated, and once the user returns the scroll to the bottom, it will again keep the scroll at the bottom. 

The overall goal of `streamContent: true` is to give the iframe a _feeling_ as if the content is being intermittently streamed in (typewriter-like effect). The reason why I say a feeling is that we are not really "streaming" content into the iframe by adding a chunk of incoming data to the end each time, we are rather rewriting the entire iframe content each time but our ways of doing it give it a stream-like effect.

Reply:
We should also distinguish that the recommended method for updating the iframe content is:

- `streamContent: false`: `api.redial(newSpec)` where the newSpec contains the new data, using the name of the iframe as the key.
- `streamContent: true`: `api.setData({ iframeName: 'new content' })` where only the data of the iframe needs to be updated.

Theres probably a better way to describe this.

Reply:
Both ways, `api.setData` should be the preferred method - why redial the entire dialog when you can just target the iframe component you want to update

Reply:
True, I realise we only do it in the plugin since the iframe component isn't initially in the dialog 🤦